bundles/binding/org.openhab.binding.mystromecopower/src/main/java/org/openhab/binding/mystromecopower/internal/api/MystromClient.java

```diff
@@ -0,0 +1,304 @@
+	 */
+	public Boolean login(){
+		Reader reader = null;
+		logger.debug("Do login for user '{}'", this.userName);
+		
+		try {
+			HttpURLConnection httpURLConnection;
+			httpURLConnection = (HttpURLConnection) new URL(API_URL + "auth?email="
+					+ this.userName
+					+ "&password=" + this.password)
+					.openConnection();
+		
+			InputStream inputStream = httpURLConnection.getInputStream();
+			reader = new InputStreamReader(inputStream, "UTF-8");
+			JsonObject jsonObject = (JsonObject) jsonParser.parse(reader);
+			
+			String status = jsonObject.get("status").getAsString();
+			if(!status.equals("ok")){
+				return false;
+			}
+			logger.debug("Logon successfull");
+			
+			authToken = jsonObject.get("authToken").getAsString();
+			
+			return true;
+		}
+		catch(Exception ex)
+		{
+			logger.error("Error do logon: '{}'", ex.toString());
+			return false;
+		} 
+		finally {
+			if (reader != null) {
+				try {
+					reader.close();
+				} catch (IOException ignored) {
+
+				}
+			}
+		}
+	}
+	
+	/* (non-Javadoc)
+	 * @see org.openhab.binding.mystromecopower.internal.api.IMystromClient#getDevices()
+	 */
+	public List<MystromDevice> getDevices()
+	{
+		Reader reader = null;
+		
+		logger.info("get devices...");
+		
+		try {
+			HttpURLConnection httpURLConnection;
+			httpURLConnection = (HttpURLConnection) new URL(API_URL + "devices" +
+					"?authToken=" + this.authToken)
+				.openConnection();
+			
+			httpURLConnection.connect();
+			
+			int responseCode = httpURLConnection.getResponseCode();
+			
+			if (responseCode != HttpURLConnection.HTTP_OK) {
+				logger.error("Get devices http code: '{}'", responseCode);
+				return new ArrayList<MystromDevice>();
+			}
+			
+			InputStream inputStream = httpURLConnection.getInputStream();
+			reader = new InputStreamReader(inputStream, "UTF-8");
+			JsonObject jsonObject = (JsonObject) jsonParser.parse(reader);
+			
+			String status = jsonObject.get("status").getAsString();
+			if(!status.equals("ok")){
+				logger.error("Error while getting devices: '{}'", status);
+				return new ArrayList<MystromDevice>();
+			}
+			
+			GetDevicesResult result = gson.fromJson(jsonObject, GetDevicesResult.class);
+			logger.info("Devices discovery sucessfull, found '{}' devices", result.devices.size());
+			
+			return result.devices;
+		}
+		catch(Exception ex){
+			logger.error("Error getting devices: '{}'", ex.toString());
+			return new ArrayList<MystromDevice>();
+		} finally {
+			if (reader != null) {
+				try {
+					reader.close();
+				} catch (IOException ignored) {
+
+				}
+			}
+		}
+	}
+	
+	/* (non-Javadoc)
+	 * @see org.openhab.binding.mystromecopower.internal.api.IMystromClient#getDeviceInfo(java.lang.String)
+	 */
+	public MystromDevice getDeviceInfo(String deviceId){
+		Reader reader = null;
+		
+		logger.debug("get device info...");
+		
+		try {
+			HttpURLConnection httpURLConnection;
+			httpURLConnection = (HttpURLConnection) new URL(API_URL + "device" +
```

Comment:
Could you please assign the URL to a local variable before opening the connection? This helps while debugging problems.